test/functional/passwords_controller_test.rb

```diff
@@ -64,6 +64,47 @@ class PasswordsControllerTest < ActionController::TestCase
 
       should "display otp form" do
         assert page.has_content?("Multi-factor authentication")
+        assert page.has_content?("OTP code")
+        assert page.has_button?("Authenticate")
+      end
+    end
+
+    context "when user has webauthn credentials but no recovery codes" do
+      setup do
+        create(:webauthn_credential, user: @user)
+        @user.mfa_recovery_codes = []
+        @user.save!
+        get :edit, params: { token: @user.confirmation_token, user_id: @user.id }
+        @controller.session[:mfa_expires_at] = 15.minutes.from_now.to_s
+      end
+
+      should respond_with :success
+
+      should "display webauthn prompt" do
+        assert page.has_button?("Authenticate with security device")
+      end
+
+      should "not display recovery code prompt" do
+        refute page.has_content?("Recovery code")
+      end
+    end
+
```

Comment:
Should there be a case with totp and webauthn enabled? (Same with the other tests)